docs/docs/using.md

```diff
@@ -0,0 +1,151 @@
+# Using JSpecify Annotations
+
+## Depending on the annotations
+
+The annotations are available from Maven Central as
+[`org.jspecify:jspecify:1.0.0`](https://repo1.maven.org/maven2/org/jspecify/jspecify/1.0.0/).
+The annotations themselves are in the `org.jspecify.annotations` package.
+
+Below are snippets you can use to add a dependency in Maven, Gradle, or Bazel.
+
+Regardless of the build system you use, avoid configuring your build to hide
+JSpecify annotations from your users. We also recommend including the
+annotations at runtime, and we have kept the JSpecify jar small to reduce the
+cost of doing so. Each build tool supports different mechanisms that will hide
+the annotation declarations, and we recommend against them in the tool-specific
+guidance below.
+
+#### Maven
+
+```xml
+<dependency>
+  <groupId>org.jspecify</groupId>
+  <artifactId>jspecify</artifactId>
+  <version>1.0.0</version>
+</dependency>
+```
+
+Avoid using [provided] or [optional] scope.
+
+#### Gradle
+
+```groovy
+dependencies {
+  api("org.jspecify:jspecify:1.0.0")
```

Comment:
This seems to only work for a `java-library` gradle plugin.
When I use this in a `java` gradle project I get `Could not find method api() for arguments [org.jspecify:jspecify:0.3.0] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.`

What is the recommendation for `java` projects? As they are not meant to be depended on, just `implementation`?

Reply:
Oh, sorry, I may at one point in the past have known that Gradle had separate plugins for "Java" and "Java Library." Thanks for pointing that out.

We need, like, a flowchart here.... I guess that's probably true of adding Gradle dependencies in general. (Maybe things also differ for Android? Hopefully [not anymore](https://github.com/google/guava/commit/3d33cd45f5d15f58b9365b63478f4d1f936f4e85).)

I am wary of presenting `implementation` as one of the top-level options, given the concerns we discuss with that when used with _libraries_. But maybe we can rework the text below?

> - We recommend avoiding [implementation] in favor of [api] whenever api is available. When implementation is the only option (such as when you use [the Java plugin](https://docs.gradle.org/current/userguide/java_plugin.html) instead of [the Java Library plugin](https://docs.gradle.org/current/userguide/java_library_plugin.html)), then it is safe to use.
> - We recommend avoiding [compileOnlyApi], though you can use it if you're confident that no users will need the annotations at runtime.

(Aside: Possibly we should use code font for `provided`, `optional`, `implementation`, and `compileOnlyApi`?)

Reply:
How's this?

Reply:
I suspect that the java-library advice also applies to [com.android.library](https://developer.android.com/studio/projects/android-library). Hopefully there aren't many such other plugins, at least common ones, so we could just mention both? (I do like that your version is shorter than mine.)

(It also occurs to me that we could try to recommend against `compileJava`, which is sort of the java equivalent to java-library's `compileOnlyApi`. But that seems less important, since users would be hurting only themselves (not their users) and since they can hopefully generalize from the java-library advice. So I don't think we need to worry about that.)

Reply:
Let's keep it simple for now. We can always add details.